ocaml/xapi/xapi_xenops.ml

```diff
@@ -2261,6 +2261,14 @@ let resync_resident_on ~__context =
       Db.VM.set_resident_on ~__context ~self:vm ~value:localhost)
       xapi_thinks_are_nowhere;
 
+  List.iter (fun ((id, state), queue_name) ->
+    let xapi_power_state =
+      xenapi_of_xenops_power_state (Some state.Vm.power_state) in
+    match xapi_power_state with
+      | `Running | `Paused -> add_caches id;
+      | `Suspended | `Halted -> ()
```

Comment:
To avoid a runtime match failure, maybe use a wildcard instead? Or introduce another explicit match `| _ -> assert false`.

Reply:
I think in this case the match is exhaustive, if I remove the second line I get a build failure complaining I forgot those 2 cases. Unfortunately `_ -> .` does not work with polymorphic variants (to ensure that the _ can never be reached).
I'll change it to `_`, its safer that way.